src/packages/hsl.tsx

```diff
@@ -0,0 +1,22 @@
+import React from "react";
+
+import ColorPicker from "../components/ColorPicker";
+import withColorModel from "../hocs/withColorModel";
+import { ColorModel, ColorPickerBaseProps, HSL } from "../types";
+import { equalColorObjects } from "../utils/compare";
+import { hslToHsv, hsvToHsl } from "../utils/convert";
+
+interface Props extends ColorPickerBaseProps<HSL> {
+  color: HSL;
+}
```

Comment:
Do we need to define `color: HSL;` here if we have `color: T;` in `ColorPickerBaseProps`?

Reply:
Seems to me that we could go with something like:
```js
const HslColorPicker: React.FC<ColorPickerBaseProps<HSL>> = withColorModel(ColorPicker, colorModel);
```

Reply:
Yeah, absolutely. With a `Partial` as the second layer in that generic though.